### PR TITLE
[CI] Run specs in random order by default

### PR DESCRIPTION
**Pull request by straight-shoota**
Running specs in random order is better than the default lexical ordering. But it's currently not the default behaviour and thus rarely used because it needs opt-in (#12540).
This patch opts into random order execution in CI and the Makefile in general.

### CODE REVIEW

**Comment by straight-shoota** [comment]
Specs are currently failing due to side effects between examples. This is fixed in #12539. These side effects have been discovered through random order execution. So this already shows the benefits =)

File: .github/workflows/openssl.yml [hunk 1]
```diff
@@ -67,4 +67,4 @@ jobs:
       - name: Check LibSSL version
         run: bin/crystal eval 'require "openssl"; p! LibSSL::OPENSSL_VERSION, LibSSL::LIBRESSL_VERSION'
```
Comment (Sija) [inline .github/workflows/openssl.yml:68]: Shouldn't this use `crystal` instead of `bin/crystal` as well?

ditto elsewhere

Comment (straight-shoota) [reply .github/workflows/openssl.yml:68]: Oh. I have no idea where those removals of `bin/` come from. Their unintended.

**Review by caspiano — APPROVED** [review]

File: Makefile [hunk 2]
```diff
@@ -26,13 +26,14 @@ junit_output ?= ## Path to output junit results
...
 check ?=        ## Enable only check when running format
+order ?=random  ## Enable order for spec execution (values: "default" | "random" | seed number)
 
 O := .build
 SOURCES := $(shell find src -name '*.cr')
 SPEC_SOURCES := $(shell find spec -name '*.cr')
 override FLAGS += -D strict_multi_assign $(if $(release),--release )$(if $(stats),--stats )$(if $(progress),--progress )$(if $(threads),--threads $(threads) )$(if $(debug),-d )$(if $(static),--static )$(if $(LDFLAGS),--link-flags="$(LDFLAGS)" )$(if $(target),--cross-compile --target $(target) )$(if $(interpreter),,-Dwithout_interpreter )
 SPEC_WARNINGS_OFF := --exclude-warnings spec/std --exclude-warnings spec/compiler --exclude-warnings spec/primitives
-SPEC_FLAGS := $(if $(verbose),-v )$(if $(junit_output),--junit_output $(junit_output) )
+SPEC_FLAGS := $(if $(verbose),-v )$(if $(junit_output),--junit_output $(junit_output) )$(if $(order),--order=$(order))
```
Comment (HertzDevil) [inline Makefile:36]:
```suggestion
SPEC_FLAGS := $(if $(verbose),-v )$(if $(junit_output),--junit_output $(junit_output) )$(if $(order),--order=$(order) )
```

ditto for Windows

Comment (straight-shoota) [reply Makefile:36]: Good catch! 👍

**Pull request merged**